crypto/x509/x509_vfy.c

```diff
@@ -3279,13 +3286,7 @@ static int build_chain(X509_STORE_CTX *ctx)
 
             x = xtmp;
             ++ctx->num_untrusted;
-            ss = cert_self_signed(ctx, xtmp);
-            if (ss < 0) {
-                X509err(X509_F_BUILD_CHAIN, ERR_R_INTERNAL_ERROR);
-                ctx->error = X509_V_ERR_UNSPECIFIED;
-                sk_X509_free(sktmp);
-                return 0;
-            }
+            self_signed = cert_self_signed(ctx, xtmp, 0) == 1;
```

Comment:
This again eliminates a check for exceptional conditions. Why?

Reply:
Yep; reverted as discussed above.